src/Xamarin.Android.Build.Tasks/Tasks/ValidateJavaVersion.cs

```diff
@@ -57,36 +54,29 @@ public override bool RunTask ()
 		//  javac 1.8.0_77
 		static readonly Regex JavacVersionRegex = new Regex (@"(?<version>[\d\.]+)(_d+)?");
 
-		bool ValidateJava (string targetFrameworkVersion, string buildToolsVersion)
+		bool ValidateJava ()
 		{
 			var java = JavaToolExe ?? (OS.IsWindows ? "java.exe" : "java");
 			var javac = JavacToolExe ?? (OS.IsWindows ? "javac.exe" : "javac");
 
-			return ValidateJava (java, JavaVersionRegex, targetFrameworkVersion, buildToolsVersion) &&
-				ValidateJava (javac, JavacVersionRegex, targetFrameworkVersion, buildToolsVersion);
+			return ValidateJava (java, JavaVersionRegex) &&
+				ValidateJava (javac, JavacVersionRegex);
 		}
 
-		bool ValidateJava (string javaExe, Regex versionRegex, string targetFrameworkVersion, string buildToolsVersion)
+		protected virtual bool ValidateJava (string javaExe, Regex versionRegex)
 		{
-			Version requiredJavaForFrameworkVersion = GetJavaVersionForFramework (targetFrameworkVersion);
-			Version requiredJavaForBuildTools = GetJavaVersionForBuildTools (buildToolsVersion);
-
-			Version required = requiredJavaForFrameworkVersion > requiredJavaForBuildTools ? requiredJavaForFrameworkVersion : requiredJavaForBuildTools;
-
+			var required = Version.Parse (MinimumSupportedJavaVersion);
 			MinimumRequiredJdkVersion = required.ToString ();
 
 			try {
 				var versionNumber = GetVersionFromTool (javaExe, versionRegex);
 				if (versionNumber != null) {
 					Log.LogMessage (MessageImportance.Normal, $"Found Java SDK version {versionNumber}.");
-					if (versionNumber < requiredJavaForFrameworkVersion) {
-						Log.LogCodedError ("XA0031", Properties.Resources.XA0031, requiredJavaForFrameworkVersion, targetFrameworkVersion);
-					}
-					if (versionNumber < requiredJavaForBuildTools) {
-						Log.LogCodedError ("XA0032", Properties.Resources.XA0032, requiredJavaForBuildTools, buildToolsVersion);
+					if (versionNumber < required) {
+						Log.LogCodedError ("XA0032", Properties.Resources.XA0031, required, "Xamarin.Android.Sdk");
```

Comment:
It looks like the message code `"XA0032"` got carried over from copy-paste and is meant to be `"XA0031"` to match the Properties.Resources.XA0031 member.